zaza/model.py

```diff
@@ -1390,8 +1390,10 @@ def f(x):
     async_block_until_services_restarted)
 
 
-async def async_block_until_unit_wl_status(unit_name, status, model_name=None,
-                                           negate_match=False, timeout=2700):
+async def async_block_until_unit_wl_status(unit_name, status,
+                                           subordinate_principal=None,
```

Comment:
Adding it to the beginning of the argument list is risky, as it's possible to use named arguments by order as well (as you do in the tests above).

Reply:
Ah, you don't do that in the tests, ignore that bit :-P

That said, it is still doable and I don't want to be the one who reviews all code that uses Zaza to see if anybody else did that :)

Reply:
I was thinking the same, it could be risky, I can change that to add to the end. You are more likely to use the subordinate_principal than the other variables, so was thinking that we could do it this way.

But agreed, this could break anything that uses this function, and adds parameters in the order that have been listed.

Reply:
Please add a doc string entry for 'subordinate_principal'

Reply:
doc string added

Reply:
In general, I agree with Chris.  If you add a default param to an call signature, it ought to go to the end to reduce potential breakage, please.